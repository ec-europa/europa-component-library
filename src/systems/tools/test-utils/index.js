const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
