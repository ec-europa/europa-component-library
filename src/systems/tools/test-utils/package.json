{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
