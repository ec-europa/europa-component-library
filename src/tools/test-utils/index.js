const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
