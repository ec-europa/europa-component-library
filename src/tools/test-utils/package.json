{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
