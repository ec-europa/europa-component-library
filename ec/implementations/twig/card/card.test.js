import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';
import dataCard from './demo/data--card';
import dataCardEvent from './demo/data--event';
import dataTile from './demo/data--tile';

describe('EC - Card', () => {
  const template = '@ecl-twig/ec-component-card/ecl-card.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(dataCard)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataCard, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataCard, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test('with validation enabled and missing input data returns the right warning message', () => {
      expect.assertions(1);

      const dataCompliance = { ...dataCard, _compliance_: true };
      dataCompliance.card.title.label = '';

      return expect(render(dataCompliance)).resolves.toMatchSnapshot();
    });
  });

  describe('Tile', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(dataTile)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataTile, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataTile, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Events', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(dataCardEvent)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataCardEvent, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataCardEvent, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
