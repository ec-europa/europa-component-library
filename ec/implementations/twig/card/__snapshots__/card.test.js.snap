// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Card Default renders correctly 1`] = `
<test>
  <article
    class="ecl-card"
  >
    <header
      class="ecl-card__header"
    >
      <div
        aria-label="card image"
        class="ecl-card__image"
        role="img"
        style="background-image:url('https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg')"
      />
      <div
        class="ecl-card__meta"
      >
        Meta1 | Meta2
      </div>
      <h1
        class="ecl-card__title"
      >
        <a
          class="ecl-link ecl-link--standalone"
          href="/example"
        >
          Better regulation
        </a>
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
    </div>
    <footer
      class="ecl-card__footer"
    >
      <ul
        class="ecl-card__info-container"
      >
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--calendar"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            2018/10/22
          </span>
        </li>
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--location"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            Luxembourg
          </span>
        </li>
      </ul>
      <ul
        class="ecl-card__tag-container"
      >
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 1
          </a>
        </li>
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 2
          </a>
        </li>
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 3
          </a>
        </li>
      </ul>
    </footer>
  </article>
</test>
`;

exports[`EC - Card Default renders correctly with extra attributes 1`] = `
<test>
  <article
    class="ecl-card"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <header
      class="ecl-card__header"
    >
      <div
        aria-label="card image"
        class="ecl-card__image"
        role="img"
        style="background-image:url('https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg')"
      />
      <div
        class="ecl-card__meta"
      >
        Meta1 | Meta2
      </div>
      <h1
        class="ecl-card__title"
      >
        <a
          class="ecl-link ecl-link--standalone"
          href="/example"
        >
          Better regulation
        </a>
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
    </div>
    <footer
      class="ecl-card__footer"
    >
      <ul
        class="ecl-card__info-container"
      >
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--calendar"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            2018/10/22
          </span>
        </li>
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--location"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            Luxembourg
          </span>
        </li>
      </ul>
      <ul
        class="ecl-card__tag-container"
      >
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 1
          </a>
        </li>
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 2
          </a>
        </li>
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 3
          </a>
        </li>
      </ul>
    </footer>
  </article>
</test>
`;

exports[`EC - Card Default renders correctly with extra class names 1`] = `
<test>
  <article
    class="ecl-card custom-class custom-class--test"
  >
    <header
      class="ecl-card__header"
    >
      <div
        aria-label="card image"
        class="ecl-card__image"
        role="img"
        style="background-image:url('https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg')"
      />
      <div
        class="ecl-card__meta"
      >
        Meta1 | Meta2
      </div>
      <h1
        class="ecl-card__title"
      >
        <a
          class="ecl-link ecl-link--standalone"
          href="/example"
        >
          Better regulation
        </a>
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
    </div>
    <footer
      class="ecl-card__footer"
    >
      <ul
        class="ecl-card__info-container"
      >
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--calendar"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            2018/10/22
          </span>
        </li>
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--location"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            Luxembourg
          </span>
        </li>
      </ul>
      <ul
        class="ecl-card__tag-container"
      >
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 1
          </a>
        </li>
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 2
          </a>
        </li>
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 3
          </a>
        </li>
      </ul>
    </footer>
  </article>
</test>
`;

exports[`EC - Card Default with validation enabled and missing input data returns the right warning message 1`] = `
<test>
  <article
    class="ecl-card"
  >
    <header
      class="ecl-card__header"
    >
      <div
        aria-label="card image"
        class="ecl-card__image"
        role="img"
        style="background-image:url('https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg')"
      />
      <div
        class="ecl-card__meta"
      >
        Meta1 | Meta2
      </div>
      <h1
        class="ecl-card__title"
      >
        <a
          class="ecl-link ecl-link--standalone"
          href="/example"
        />
        <!-- ecl2-compliance: link--standalone -->
        <div
          class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
        >
          <p
            class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
          >
            [link--standalone] - {{ link.label }} and none of {{ link.icon_position }}, {{ icon.name }}, {{ icon.path }} or {{ icon.type }} can be empty at the same time, an unusable link would be generated otherwise
          </p>
        </div>
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
    </div>
    <footer
      class="ecl-card__footer"
    >
      <ul
        class="ecl-card__info-container"
      >
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--calendar"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
          <span
            class="ecl-card__info-label"
          >
            2018/10/22
          </span>
        </li>
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--location"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
          <span
            class="ecl-card__info-label"
          >
            Luxembourg
          </span>
        </li>
      </ul>
      <ul
        class="ecl-card__tag-container"
      >
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 1
          </a>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[tag--link] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
        </li>
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 2
          </a>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[tag--link] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
        </li>
        <li
          class="ecl-card__tag-item"
        >
          <a
            class="ecl-tag ecl-card__tag"
            href="/example"
          >
            tag 3
          </a>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[tag--link] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
        </li>
      </ul>
    </footer>
  </article>
  <!-- ecl2-compliance: card -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
    >
      [card] -  {{ card.title.label }} property needs to set and filled in.
    </p>
  </div>
</test>
`;

exports[`EC - Card Events renders correctly 1`] = `
<test>
  <article
    class="ecl-card"
  >
    <header
      class="ecl-card__header"
    >
      <div
        aria-label="card image"
        class="ecl-card__image"
        role="img"
        style="background-image:url('https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg')"
      />
      <h1
        class="ecl-card__title"
      >
        <a
          class="ecl-link ecl-link--standalone"
          href="/example"
        >
          Better regulation
        </a>
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
    </div>
    <footer
      class="ecl-card__footer"
    >
      <ul
        class="ecl-card__info-container"
      >
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--calendar"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            2018/10/22
          </span>
        </li>
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--location"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            Luxembourg
          </span>
        </li>
      </ul>
    </footer>
  </article>
</test>
`;

exports[`EC - Card Events renders correctly with extra attributes 1`] = `
<test>
  <article
    class="ecl-card"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <header
      class="ecl-card__header"
    >
      <div
        aria-label="card image"
        class="ecl-card__image"
        role="img"
        style="background-image:url('https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg')"
      />
      <h1
        class="ecl-card__title"
      >
        <a
          class="ecl-link ecl-link--standalone"
          href="/example"
        >
          Better regulation
        </a>
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
    </div>
    <footer
      class="ecl-card__footer"
    >
      <ul
        class="ecl-card__info-container"
      >
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--calendar"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            2018/10/22
          </span>
        </li>
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--location"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            Luxembourg
          </span>
        </li>
      </ul>
    </footer>
  </article>
</test>
`;

exports[`EC - Card Events renders correctly with extra class names 1`] = `
<test>
  <article
    class="ecl-card custom-class custom-class--test"
  >
    <header
      class="ecl-card__header"
    >
      <div
        aria-label="card image"
        class="ecl-card__image"
        role="img"
        style="background-image:url('https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg')"
      />
      <h1
        class="ecl-card__title"
      >
        <a
          class="ecl-link ecl-link--standalone"
          href="/example"
        >
          Better regulation
        </a>
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
    </div>
    <footer
      class="ecl-card__footer"
    >
      <ul
        class="ecl-card__info-container"
      >
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--calendar"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            2018/10/22
          </span>
        </li>
        <li
          class="ecl-card__info-item"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#general--location"
            />
          </svg>
          <span
            class="ecl-card__info-label"
          >
            Luxembourg
          </span>
        </li>
      </ul>
    </footer>
  </article>
</test>
`;

exports[`EC - Card Tile renders correctly 1`] = `
<test>
  <article
    class="ecl-card ecl-card--tile"
  >
    <header
      class="ecl-card__header"
    >
      <h1
        class="ecl-card__title"
      >
        Better regulation
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
      <ul
        class="ecl-card__link-container"
      >
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 1
          </a>
        </li>
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 2
          </a>
        </li>
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 3
          </a>
        </li>
      </ul>
    </div>
    <footer
      class="ecl-card__footer"
    />
  </article>
</test>
`;

exports[`EC - Card Tile renders correctly with extra attributes 1`] = `
<test>
  <article
    class="ecl-card ecl-card--tile"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <header
      class="ecl-card__header"
    >
      <h1
        class="ecl-card__title"
      >
        Better regulation
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
      <ul
        class="ecl-card__link-container"
      >
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 1
          </a>
        </li>
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 2
          </a>
        </li>
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 3
          </a>
        </li>
      </ul>
    </div>
    <footer
      class="ecl-card__footer"
    />
  </article>
</test>
`;

exports[`EC - Card Tile renders correctly with extra class names 1`] = `
<test>
  <article
    class="ecl-card ecl-card--tile custom-class custom-class--test"
  >
    <header
      class="ecl-card__header"
    >
      <h1
        class="ecl-card__title"
      >
        Better regulation
      </h1>
    </header>
    <div
      class="ecl-card__body"
    >
      <div
        class="ecl-card__description"
      >
        Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
      </div>
      <ul
        class="ecl-card__link-container"
      >
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 1
          </a>
        </li>
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 2
          </a>
        </li>
        <li
          class="ecl-card__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-card__link"
            href="/example"
          >
            link 3
          </a>
        </li>
      </ul>
    </div>
    <footer
      class="ecl-card__footer"
    />
  </article>
</test>
`;
